presto-hive-functions/src/main/java/com/facebook/presto/hive/functions/gen/CompilerUtils.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.hive.functions.gen;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.bytecode.ClassDefinition;
+import com.facebook.presto.bytecode.DynamicClassLoader;
+import com.facebook.presto.bytecode.ParameterizedType;
+
+import java.io.PrintWriter;
+import java.lang.invoke.MethodHandle;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.facebook.presto.bytecode.BytecodeUtils.toJavaIdentifierString;
+import static com.facebook.presto.bytecode.ClassGenerator.classGenerator;
+import static com.facebook.presto.bytecode.ParameterizedType.typeFromJavaClassName;
+import static java.time.ZoneOffset.UTC;
+
+public final class CompilerUtils
```

Comment:
These are just copied from `compilerUtils` in presto-main right? Might consider move that to presto-bytecode if useful. Just for consideration.